ext/standard/basic_functions.stub.php

```diff
@@ -395,3 +395,199 @@ function assert($assertion, $description = null): bool {}
+function strrchr(string $haystack, string $needle) {}
+
+function chunk_split(string $str, int $chunklen = 76, string $ending = "\r\n"): string {}
+
+/** @return string|false */
+function substr(string $str, int $start, int $length = UNKNOWN) {}
+
+/**
+ * @param mixed $str
+ * @param mixed $replace
+ * @param mixed $start
+ * @param mixed $length
+ * @return string|array|false
+ */
+function substr_replace($str, $replace, $start, $length = UNKNOWN) {}
+
+function quotemeta(string $str): string {}
+
+function ord(string $character): int {}
+
+function chr(int $codepoint): string {}
+
+function ucfirst(string $str): string {}
+
+function lcfirst(string $str): string {}
+
+function ucwords(string $str, string $delimiters = " \t\r\n\f\v"): string {}
+
+/** @param string|array $from */
+function strtr(string $str, $from, string $to = UNKNOWN): string {}
+
+function strrev(string $str): string {}
+
+function similar_text(string $str1, string $str2, float &$percent = UNKNOWN): int {}
+
+function addcslashes(string $str, string $charlist): string {}
+
+function addslashes(string $str): string {}
+
+function stripcslashes(string $str): string {}
+
+function stripslashes(string $str): string {}
+
+/**
+ * @param string|array $search
+ * @param string|array $replace
+ * @param string|array $subject
+ * @param int $replace_count
+ * @return string|array
+ */
+function str_replace($search, $replace, $subject, &$replace_count = UNKNOWN) {}
+
+/**
+ * @param string|array $search
+ * @param string|array $replace
+ * @param string|array $subject
+ * @param int $replace_count
+ * @return string|array
+ */
+function str_ireplace($search, $replace, $subject, &$replace_count = UNKNOWN) {}
+
+function hebrev(string $str, int $max_chars_per_line = 0): string {}
+
+function hebrevc(string $str, int $max_chars_per_line = 0): string {}
+
+function nl2br(string $str, bool $is_xhtml = true): string {}
+
+/** @param mixed $allowable_tags */
+function strip_tags(string $str, $allowable_tags = UNKNOWN): string {}
+
+/**
+ * @param string|array $locales
+ * @return string|false
+ */
+function setlocale(int $category, $locales, string ...$rest) {}
+
+/** @param array $result */
+function parse_str(string $encoded_string, &$result): void {}
+
+function str_getcsv(string $string, string $delimiter = ',', string $enclosure = '"', string $escape = '\\'): array {}
+
+function str_repeat(string $input, int $mult): string {}
+
+function count_chars(string $input, int $mode = 0) {}
+
+function strnatcmp(string $s1, string $s2): int {}
+
+function localeconv(): array {}
+
+function strnatcasecmp(string $s1, string $s2): int {}
+
+/** @return int|false */
+function substr_count(string $haystack, string $needle, int $offset = 0, int $length = UNKNOWN) {}
+
+function str_pad(string $input, int $pad_length, string $pad_string = " ", int $pad_type = STR_PAD_RIGHT): string {}
+
+/** @return array|int */
+function sscanf(string $str, string $format, string &...$vars) {}
```

Comment:
```suggestion
function sscanf(string $str, string $format, &...$vars) {}
```

Even as an output type, `string` wouldn't be right here.